src/lib/json-ld.js

```diff
@@ -0,0 +1,138 @@
+import { Helmet } from 'react-helmet';
+
+import { authorPathByName } from 'lib/users';
+import { postPathBySlug } from 'lib/posts';
+import { pagePathBySlug } from 'lib/pages';
+
+import config from '../../package.json';
+
+export function ArticleJsonLd({ post = {}, siteTitle = '' }) {
+  const { homepage = '', faviconPath = '/favicon.ico' } = config;
+  const { title, slug, excerpt, date, author, categories, modified, featuredImage } = post;
+  const path = postPathBySlug(slug);
+  const datePublished = !!date && new Date(date);
+  const dateModified = !!modified && new Date(modified);
+
+  /** TODO - As image is a recommended field would be interesting to have a
+   * default image in case there is no featuredImage comming from WP,
+   * like the open graph social image
+   * */
+
+  const jsonLd = {
+    '@context': 'https://schema.org',
+    '@type': 'Article',
+    mainEntityOfPage: {
+      '@type': 'WebPage',
+      '@id': `${homepage}${path}`,
+    },
+    headline: title,
+    image: [featuredImage?.sourceUrl],
+    datePublished: datePublished.toISOString(),
+    dateModified: dateModified.toISOString() || datePublished.toISOString(),
```

Comment:
wouldn't this still error if `dateModified` is undefined?

maybe we would want `dateModified ? dateModified.toISOString() : datePublished.toISOString()`?

Reply:
Totally right, if **date** or **modified** comes with `undefined` is trying to do `false`.toISOString()  :sweat_smile: . I'll change, sorry about that.